presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -150,8 +170,100 @@ public void testColumnsCached()
         assertThat(cachingJdbcClient.getColumns(SESSION, table)).contains(phantomColumn);
         jdbcClient.dropColumn(SESSION, table, phantomColumn);
 
+        // Load column into cache
         assertThat(jdbcClient.getColumns(SESSION, table)).doesNotContain(phantomColumn);
         assertThat(cachingJdbcClient.getColumns(SESSION, table)).contains(phantomColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 1, 1, 1);
+
+        // Read column from cache
+        assertThat(cachingJdbcClient.getColumns(SESSION, table)).contains(phantomColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 1, 2, 1);
+    }
+
+    @Test
+    public void testColumnsCachedBySession()
+    {
+        ConnectorSession firstSession = createSession("first");
+        ConnectorSession secondSession = createSession("second");
+        JdbcTableHandle table = getAnyTable(schema);
+
+        JdbcColumnHandle phantomColumn = addColumn(table);
+
+        // Load columns in first session scope
+        assertThat(cachingJdbcClient.getColumns(firstSession, table)).contains(phantomColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 1, 0, 1);
+
+        // Load columns in second session scope
+        assertThat(cachingJdbcClient.getColumns(secondSession, table)).contains(phantomColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 2, 0, 2);
+
+        // Drop first column and invalidate cache for all sessions
```

Comment:
maybe add check here that something is cached before invalidation?

Reply:
Done